app/controllers/users/chat_rooms/messages_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Users::ChatRooms::MessagesController < ApplicationController
+  before_action :authenticate_user!
+  
+  def index
+    @chat_room = ChatRoom.find(params[:chat_room_id])
+    @messages = Message.where(chat_room_id: @chat_room.id)
```

Comment:
chat_roomに紐づいたmessageはアソシエーションで取得できるので、アソシエーションで取得しましょう。

アソシエーションで取得すると、間違えたchat_roomからのメッセージを取得する間違いを起こしにくくなります。

```suggestion
    @messages = @chat_room.messages
```